components/servo/generic/servo.go

```diff
@@ -0,0 +1,345 @@
+// Package servogeneric implements a generic servo
+package servogeneric
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	viamutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/servo"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/utils"
+)
+
+const (
+	minDeg     float64 = 0.0
+	maxDeg     float64 = 180.0
+	minWidthUs uint    = 500  // absolute minimum width pwm width
+	maxWidthUs uint    = 2500 // absolute maximum width pwm width
+)
+
+type servoConfig struct {
+	Pin   string `json:"pin"`   // Pin a GPIO pin with pwm capabilities
+	Board string `json:"board"` // Board a board that exposes GPIO pins
+	// MinDeg minimum angle the servo can reach, note this doesn't affect PWM calculation
+	MinDeg *float64 `json:"min_angle_deg,omitempty"`
+	// MaxDeg maximum angle the servo can reach, note this doesn't affect PWM calculation
+	MaxDeg *float64 `json:"max_angle_deg,omitempty"`
```

Comment:
It's a neat trick to avoid defining default value, for example if it wasn't a pointer and a user didn't reference this field it would be initialized to 0.0 by default. Now for minDeg it's fine but for max we have an issue :) Also we consume the config struct and never reference it again, and we can reasonably assume that nothing is going to change anything while we read it